WordPress/src/androidTest/java/org/wordpress/android/util/HealthCheckTest.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    private void runUrlCanonicalization(String testCaseComment, JSONObject testSetup) throws JSONException {
+        final JSONObject input = testSetup.getJSONObject("input");
+
+        final String inputUrl = input.isNull("siteUrl") ? null : input.getString("siteUrl");
+
+        final JSONObject output = testSetup.getJSONObject("output");
+
+        final String outputUrl = output.optString("siteUrl", null);
+        final JSONObject error = output.optJSONObject("error");
+
+        String canonicalizedUrl = null;
+        try {
+            canonicalizedUrl = XMLRPCUtils.sanitizeSiteUrl(inputUrl);
+
+            // if we reached this point, it means that no error occurred
+            assertNull(testCaseMessage("Testcase defines an error but no error occurred!", testCaseComment), error);
+        } catch (XMLRPCUtils.XMLRPCUtilsException hce) {
+            assertNotNull(testCaseMessage("Error occurred but testcase does not define an error!", testCaseComment),
+                    error);
+
+            assertEquals(testCaseMessage("Error message does not match the defined one!", testCaseComment), error
+                    .getString("message"), getInstrumentation().getTargetContext().getString(hce.errorMsgId));
+        }
+
+        assertEquals(testCaseMessage("Canonicalized URL does not match the defined one!", testCaseComment),
+                outputUrl, canonicalizedUrl);
+    }
+
+    private void runXmlrpcDiscovery(String testCaseComment, JSONObject testSetup) throws JSONException, IOException {
+        final MockWebServer server = new MockWebServer();
+
+        testSetup = new JSONObject(replaceServerMagicName(server, testSetup.toString()));
+
+        final JSONObject input = testSetup.getJSONObject("input");
+
+        setupMockHttpServer(server, input);
+
+        final String inputUrl = input.isNull("siteUrl") ? server.url("").toString() : input.getString("siteUrl");
+
+        final JSONObject output = testSetup.getJSONObject("output");
+
+        final String outputUrl = output.optString("xmlrpcEndpoint", null);
+        final JSONObject error = output.optJSONObject("error");
+
+        String xmlrpcUrl = null;
+        try {
+            xmlrpcUrl = XMLRPCUtils.verifyOrDiscoverXmlRpcUrl(inputUrl, input.optString("username", null), input
+                    .optString("username", null));
+
+            // if we reached this point, it means that no error occurred
+            assertNull(testCaseMessage("Testcase defines an error but no error occurred!", testCaseComment), error);
+        } catch (XMLRPCUtils.XMLRPCUtilsException hce) {
+            assertNotNull(testCaseMessage("Error occurred but testcase does not define an error!", testCaseComment),
+                    error);
+
+            assertEquals(testCaseMessage("Error message does not match the defined one!", testCaseComment), error
+                    .getString("message"), getInstrumentation().getTargetContext().getString(hce.errorMsgId));
```

Comment:
Looks like `errorMsgId` can't be compared to the predefined message here because of i18n consideration:
```
org.wordpress.android.util.HealthCheckTest > testHealthCheckXplat
FAILED 
	junit.framework.ComparisonFailure: Error message does not match the defined one! (on testCase: 'testing xmlrpc.php missing') expected:<[Couldn't connect to the WordPress site]> but was:<[Αδυναμία σύνδεσης στον ιστότοπο του WordPress]>
	at org.wordpress.android.util.HealthCheckTest.runXmlrpcDiscovery(HealthCheckTest.java:130)
```